Source/WebKit/UIProcess/API/Cocoa/NSAttributedString.mm

```diff
@@ -164,6 +184,45 @@ + (void)clearConfiguration
     globalConfiguration() = nil;
 }
 
++ (void)validateEntry:(id)maybeFileURL
+{
+    NSString *errorMessage = nil;
+
+    auto* url = dynamic_objc_cast<NSURL>(maybeFileURL);
+    if (!url)
+        errorMessage = @"The NSDictionary associated with _WKReadAccessFileURLsOption may only contain NSURL objects.";
+    else if (!url.fileURL)
+        errorMessage = @"_WKReadAccessFileURLsOption requires its NSURL objects to be file URLs.";
+
+    if (!errorMessage)
+        return;
+
+    if (self.readOnlyAccessPaths.get().count) {
+        [self.readOnlyAccessPaths removeAllObjects];
+        [self clearConfiguration];
+    }
+    [NSException raise:NSInvalidArgumentException format:@"%@", errorMessage];
+}
+
++ (void)maybeConsumeBundlePaths:(NSDictionary<NSAttributedStringDocumentReadingOptionKey, id> *)options
+{
+    auto* readAccessFileURLs = dynamic_objc_cast<NSArray<NSURL *>>(options[_WKReadAccessFileURLsOption]);
+    if (readAccessFileURLs.count > maximumReadOnlyAccessPaths)
+        [NSException raise:NSInvalidArgumentException format:@"_WKReadAccessFileURLsOption may have at most two additional directories."];
+
+    for (id fileURL in readAccessFileURLs)
+        [self validateEntry:fileURL];
+
+    if ([readAccessFileURLs isEqualToArray:self.readOnlyAccessPaths.get()])
+        return;
+
+    if (readAccessFileURLs)
+        [self.readOnlyAccessPaths setArray:readAccessFileURLs];
+    else
+        [self.readOnlyAccessPaths removeAllObjects];
```

Comment:
Does this work, or does it just remove the objects from a copy of the array? I think this might need to be `[self.readOnlyAccessPaths setArray:@[]]`.

Reply:
Ditto the above. I will confirm.